src/mailkit.cs

```diff
@@ -0,0 +1,423 @@
+		NSData RawData { get; }
+
+		[Export ("isSigned")]
+		bool IsSigned { get; }
+
+		[Export ("isEncrypted")]
+		bool IsEncrypted { get; }
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEEmailAddress : NSSecureCoding, NSCopying
+	{
+		[Export ("initWithRawString:")]
+		IntPtr Constructor (string rawString);
+
+		[Export ("rawString")]
+		string RawString { get; }
+
+		[NullAllowed, Export ("addressString")]
+		string AddressString { get; }
+	}
+
+	delegate void SessionHandler (NSDictionary<MEEmailAddress, MEAddressAnnotation> sessionDictionary);
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Protocol, Advice ("'MEComposeSessionHandler' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	interface MEComposeSessionHandler
+	{
+		[Abstract]
+		[Export ("mailComposeSessionDidBegin:")]
+		void MailComposeSessionDidBegin (MEComposeSession session);
+
+		[Abstract]
+		[Export ("mailComposeSessionDidEnd:")]
+		void MailComposeSessionDidEnd (MEComposeSession session);
+
+		[Abstract]
+		[Export ("viewControllerForSession:")]
+		MEExtensionViewController GetViewController (MEComposeSession session);
+
+		[Async]
+		[Export ("session:annotateAddressesWithCompletionHandler:")]
+		void AnnotateAddress (MEComposeSession session, SessionHandler completionHandler);
+
+		[Async]
+		[Export ("session:canSendMessageWithCompletionHandler:")]
+		void AllowMessageSend (MEComposeSession session, Action<NSError> completionHandler);
+
+		[Export ("additionalHeadersForSession:")]
+		NSDictionary<NSString, NSArray<NSString>> GetAdditionalHeaders (MEComposeSession session);
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Protocol, Advice ("'MEContentBlocker' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	interface MEContentBlocker
+	{
+		[Abstract]
+		[Export ("contentRulesJSON")]
+		NSData ContentRulesJson { get; }
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Protocol, Advice ("'MEExtension' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	interface MEExtension
+	{
+		[Export ("handlerForComposeSession:")]
+		MEComposeSessionHandler HandlerForComposeSession (MEComposeSession session);
+
+		[Export ("handlerForMessageActions")]
+		MEMessageActionHandler HandlerForMessageActions { get; }
+
+		[Export ("handlerForContentBlocker")]
+		MEContentBlocker HandlerForContentBlocker { get; }
+
+		[Export ("handlerForMessageSecurity")]
+		IMEMessageSecurityHandler HandlerForMessageSecurity { get; }
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Protocol, Advice ("'MEMessageActionHandler' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	interface MEMessageActionHandler
+	{
+		[Abstract]
+		[Export ("decideActionForMessage:completionHandler:")]
+		void CompletionHandler (MEMessage message, Action<MEMessageActionDecision> completionHandler);
```

Comment:
wrong name, it's the `decideAction` that's the important part of the selector name
`ForMessage:completionHandler:` are the parameters